test/HealthChecks.Qdrant.Tests/Functional/QdrantHealthCheckTests.cs

```diff
@@ -0,0 +1,85 @@
+using System.Net;
+using Qdrant.Client;
+
+namespace HealthChecks.Qdrant.Tests.Functional;
+
+public class qdrant_healthcheck_should
+{
+    [Fact]
+    public async Task be_healthy_when_qdrant_is_available_using_client_factory()
+    {
+        var webHostBuilder = new WebHostBuilder()
+            .ConfigureServices(services =>
+            {
+                services
+                    .AddHealthChecks()
+                    .AddQdrant(
+                        clientFactory: sp => new QdrantClient("localhost"), tags: new string[] { "qdrant" });
+            })
+            .Configure(app =>
+            {
+                app.UseHealthChecks("/health", new HealthCheckOptions
+                {
+                    Predicate = r => r.Tags.Contains("qdrant")
+                });
+            });
+
+        using var server = new TestServer(webHostBuilder);
+
+        using var response = await server.CreateRequest("/health").GetAsync();
+
+        response.StatusCode.ShouldBe(HttpStatusCode.OK);
```

Comment:
For my learning, how does this return OK when talking to the localhost? There isn't a Qdrant instance running, is there?

Reply:
Qdrant instance running by ci action
see: 
https://github.com/Xabaril/AspNetCore.Diagnostics.HealthChecks/pull/2310/files#diff-5d6b2fd45c59f6628e08c23dcf560e4b79d0a6484e47de1a258a58d063d2f21aR31-R36

In most cases, we use this approach to run services in this repo.

Reply:
But if I just run the test from Test Explorer, it fails, right?

Reply:
I think so.